src/mock/mocker.spec.ts

```diff
@@ -228,3 +232,252 @@ test('Multiple Mockers with their MockContact(s)', async t => {
   t.notEqual(contact1, contact2, 'should have seperate MockContact classes')
   t.equal(contact1.id, contact2.id, 'should have the same id from different mocker instances')
 })
+
+test('MockContact.say(url).to(contact)', async t => {
+  const {
+    user,
+    mary,
+    puppet,
+  }         = createFixture()
+
+  const url = new UrlLink({
+    title: 'mock',
+    url: 'http://www.mock.com',
+  })
+
+  await puppet.start()
+
+  let receive
+  mary.on('message', async message => {
+    receive = await message.toUrlLink()
+  })
+
+  user.say(url).to(mary)
+
+  await sleep(1000)
```

Comment:
Please do not use `sleep()` in the unit testings until you are time mock tools like [Fake timers](https://sinonjs.org/releases/v9.0.2/fake-timers/) at the same time, because this will slow down the unit test.

I'd like to suggest that you can use a `await new Promise(setImmediate)` at here, to make sure we will clean the micro tasks in the background before we continue to run the blow codes.

Reply:
time faker has some problem, all tests use the same clock when go to the next 1000ms tick, other test cases will fail.
I don't know if my usage is correct.
```typescript

  let receive
  mary.on('message', async message => {
    receive = await message.toUrlLink()
    // skip clock
    skipTick(3000)
  })

  user.say(url).to(mary)
  // make sleep
  await sleep(3000)
  t.deepEqual(receive, url, 'should received the expected contact message payload')
  await puppet.stop()
```

Reply:
use setImmediate it works fine.